src/main/java/com/google/sps/servlets/ListDrugsServlet.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main.java.com.google.sps.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.gson.Gson;
+import main.java.com.google.sps.data.Drug;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for listing comments. */
+@WebServlet("/list-drugs")
+public class ListDrugsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    Query<Entity> query =
+        Query.newEntityQueryBuilder().setKind("Drug").setOrderBy(OrderBy.asc("name")).build();
+    QueryResults<Entity> results = datastore.run(query);
+
+    List<Drug> drugs = new ArrayList<>();
+    while (results.hasNext()) {
+      Entity entity = results.next();
+
+      long id = entity.getKey().getId();
+      String name = entity.getString("name");
+      String whatIs = entity.getString("whatIs");
```

Comment:
This is a nice list of potential properties. I would probably keep list-drugs returning the name, id and aliases and then have a get-drug which returns all of the details for a single drug. This servlet would be called when we need a list of drugs by name or alias. The returned ID can be passed as a parameter to the get-drug servlet which can use datastore.get(keyFactory.newKey(id)). 

See walkthroughs/week-3-libraries/datastore/examples/todo-list/src/main/java/com/google/sps/servlets/DeleteTaskServlet.java for an example of looking up an entity by id.